static/app/views/explore/logs/fieldRenderers.tsx

```diff
@@ -158,26 +163,94 @@ function CodePathRenderer(props: LogFieldRendererProps) {
     orgSlug: props.extra.organization.slug,
     projectSlug: props.extra.projectSlug ?? '',
     releaseVersion: typeof releaseVersion === 'string' ? releaseVersion : '',
+    enabled: shouldLoad,
   });
-  const {data: codeLink} = useStacktraceLink({
-    event: {
-      release,
-      sdk,
-    },
-    frame: {
-      function: typeof codeFunctionName === 'string' ? codeFunctionName : undefined,
-      lineNo: codeLineNumber ? +codeLineNumber : undefined,
-      filename: typeof filename === 'string' ? filename : undefined,
+  const {data: codeLink} = useStacktraceLink(
+    {
+      event: {
+        release,
+        sdk,
+      },
+      frame: {
+        function: typeof codeFunctionName === 'string' ? codeFunctionName : undefined,
+        lineNo: codeLineNumber ? +codeLineNumber : undefined,
+        filename: typeof filename === 'string' ? filename : undefined,
+      },
+      orgSlug: props.extra.organization.slug,
+      projectSlug: props.extra.projectSlug ?? '',
     },
-    orgSlug: props.extra.organization.slug,
-    projectSlug: props.extra.projectSlug ?? '',
-  });
+    {
+      enabled: shouldLoad && !!props.extra.projectSlug && !!release,
+    }
+  );
 
-  if (codeLink?.sourceUrl) {
-    return <Link to={codeLink.sourceUrl}>{props.basicRendered}</Link>;
-  }
+  return (
+    <span data-test-id="hoverable-code-path" {...wrapperProps}>
+      {codeLink?.sourceUrl ? (
+        <Link data-test-id="hoverable-code-path-link" to={codeLink.sourceUrl}>
+          {props.basicRendered}
+        </Link>
+      ) : (
+        props.basicRendered
+      )}
+    </span>
+  );
+}
+
+/**
+ * This hook is used to lazy load an attribute on hover.
+ * This is used to avoid rendering extra content (eg. making api calls) for constantly shown fields (eg. adding a column in the table view).
+ */
+function useLazyLoadAttributeOnHover(hoverTimeout: number, props: LogFieldRendererProps) {
+  const [shouldLoad, setShouldLoad] = useState(props.extra.disableLazyLoad === true);
+  const hoverTimeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+  const handleMouseEnter = useCallback(() => {
+    if (shouldLoad) {
+      return () => {};
+    }
+
+    if (hoverTimeoutRef.current) {
+      clearTimeout(hoverTimeoutRef.current);
+    }
+
+    hoverTimeoutRef.current = setTimeout(() => {
+      setShouldLoad(true);
+    }, hoverTimeout);
 
-  return props.basicRendered;
+    return () => {
+      if (hoverTimeoutRef.current) {
+        clearTimeout(hoverTimeoutRef.current);
+      }
+    };
+  }, [hoverTimeout, shouldLoad]);
+
+  const handleMouseLeave = useCallback(() => {
+    if (shouldLoad) {
+      return;
+    }
+
```

Comment:
This check seems unnecessary. Feels like you can just always clear the timeout if one exists on mouse leave.